### DISCUSSION TITLE
Exclude from running gomplate but still be part of the output

### DISCUSSION THREAD

**Discussion by vivekragunathan** [hairyhenderson/gomplate#1884] [Q&A] [answered]
Is there a way to tell gomplate to exclude certain files from running the template on but still be part of the output folder?

The case I am dealing with is `*.jar` files that need to be part of the output folder but gomplate should not run templates on jar files. If I specify `—exclude` option, the jar files are not part of the output folder (and the template is not run on them).

Is there a way to achieve this? Thanks. 🙏

**Comment by assayire** [comment]
@hairyhenderson Is the above possible to achieve?

**Comment by hairyhenderson** [comment]
Currently this is not possible directly... One way to achieve this could be to call `cp` as a postExec command: https://docs.gomplate.ca/config/#postexec

But given that gomplate is a templating tool, not a general file-copying tool, I'm not sure it makes sense to support this sort of functionality...

**Reply by assayire** [reply]
I am not sure if I explained it right. The task here is not just copying which gomplate already does when running the command. `gomplate` processes entire directory of files and sub-directories, and copies/creates/materializes each one in a desired target folder. The problem is when gomplate tries to process files like `.jar` or images and fails because of unexpected delimiter like characters in those files. This is totally not predictable because we cannot say what those binary files contain and if they happen to overlap with gomplate or any other templating engine syntax.

It would be great if gomplate could support exclude such files from processing for templating but still copy them to the target folder. This is a classic case when the folder you are processing is a project template (like Java or other).

Exclusion may fall into one of the following categories

- `None` (default)
- `FromProcessing` (but shall be copied to the target folder)
- `FromCopying` (+Processing, the —exclude option)

Something like `—exclude-processing` option would be great.

Manually copying using the `postexec` is possible but fragile because it is hard to guarantee that _all_ the files not processed are picked up without skipping any file or folder and copied in the `postexec` step.

**Reply by hairyhenderson** [reply]
> I am not sure if I explained it right.
er, you didn't explain it at all, unless you're the same person as @vivekragunathan 😅

Thanks for clarifying though, I think I understand better what you're trying to achieve.

Can you open a new issue for this new feature?

**Reply by assayire** [reply, marked as answer]
Done https://github.com/hairyhenderson/gomplate/issues/1911
